### PR TITLE
For updating cache periodically

### PR DESCRIPTION
**Pull request by jaydhulia**
- [x] Update cache on a timer based on config settings
- [x] Enable/Disable enabling of caching
- [x] Update Scenario cache
- [x] Update Scene cache
- [x] Update asset cache

### CODE REVIEW

File: app/library/postgres.py [hunk 1]
```diff
@@ -34,15 +34,15 @@ async def post_asset_to_postgres(data: dict, session):
     return asset_model.as_dict()
 
 
-async def get_object_from_postgres(object_id: str, session):
+async def get_object_from_postgres(object_id: str, session, update_cache=False):
```
Comment (drbalar) [inline app/library/postgres.py:37]: `update_cache` is always `True` regardless of the config value.. I'm guessing that is going to be implemented in another PR?

Comment (ahmedhamodi) [reply app/library/postgres.py:37]: Did you mean always false? and I think this is a default value for if the field isn't set

Comment (drbalar) [reply app/library/postgres.py:37]: The function that calls `get_object_from_postgres` is always passing in true

Comment (ahmedhamodi) [reply app/library/postgres.py:37]: Ah I see, lines 21-25 in update_cache.py. Good point

File: configs/dev-config.yaml [hunk 2]
```diff
@@ -30,3 +30,7 @@ script:
 cache:
   # seconds in day: 24 * 60 * 60
   expire_time: 86400
+  # how frequently to update cache (in seconds), 60 seconds for dev, for now
+  update_time: 5
```
[the remark this answers is not in the document]
Comment (jaydhulia) [reply configs/dev-config.yaml:34]: Yup, this should be 60 if we actually enable it, but since we aren't enabling it for dev, it doesn't matter 😄

File: app/cache/update_cache.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+from cache.cache import clear_all_cache, update_scenario_cache
+from library.postgres import get_scene_from_postgres
+from models import get_session
+from models.db_client import get_scenarios
+
+
+async def update_cache():
+    session = get_session()
+    clear_all_cache()
```
[the remark this answers is not in the document]
Comment (jaydhulia) [reply app/cache/update_cache.py:12]: Ah good catch, will update

File: app/library/postgres.py (hunk 1 above)
Comment (jaydhulia) [reply app/library/postgres.py:37]: Nope this is meant to be always true in that function.
The config's `update_enabled` being `True` or `False` determines whether this function runs or not. 
If this function runs, the `update_cache` parameter should always be true, otherwise there is no point in running this function 😄 Does that make sense?

File: app/cache/update_cache.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jaydhulia) [reply app/cache/update_cache.py]: Updated!

File: app/library/postgres.py (hunk 1 above)
Comment (jaydhulia) [reply app/library/postgres.py:37]: An example of where `update_cache` would be `False` is the admin routes. For the user routes, and the update_cache.py we always want it to be true, so that it updates the cache. The logic for the config's `update_enabled` being `True` or `False` is handled in main (https://github.com/uwblueprint/cdc-backend/pull/65/files#diff-a4bc3d28ebe484212cebe359972290c5dd689f559c1dda8d6f71014815f313d1R109)

Comment (drbalar) [reply app/library/postgres.py:37]: Ahhh got it. So the config's `update_enabled` only affects the admin routes, not the user routes. That makes sense, ty

**Review by drbalar — APPROVED** [review]
LG

File: app/library/postgres.py (hunk 1 above)
Comment (jaydhulia) [reply app/library/postgres.py:37]: Well no, sorry, to clarify:
- config's `update_enabled` determines whether calling the `update_cache` method periodically is enabled or not.
- the parameter `update_cache` is always False for admin routes, and always True for user routes, independent of config's `update_enabled`

Does that make sense?

Comment (drbalar) [reply app/library/postgres.py:37]: Oh got it now. The naming tripped me lol

Comment (jaydhulia) [reply app/library/postgres.py:37]: Fair enough, I will update the config's name to `periodic_callback_enabled`

Comment (jaydhulia) [reply app/library/postgres.py:37]: Updated, please re-approve when you get a chance, ty!

**Review by drbalar — APPROVED** [review]
thank you :D
